purepyvmware/datastores.py

```diff
@@ -0,0 +1,329 @@
+
+    def verify_vsphere_cluster(self, cluster_name, flasharray):
+        """Verify host objects and host groups associated with the vSphere Cluster are configured on the FlashArray.
+
+        Args:
+            cluster_name (str): Name of the vSphere Cluster to verify.
+            flasharray (purestorage.FlashArray): Existing session to FlashArray.
+
+        Returns:
+            hgroup (dict): Host group associated with vSphere cluster.
+            connected_esxi_hosts (list): List of ESXi hosts in a connected state for vSphere cluster..
+        """
+        msg = 'Cancelling datastore creation. Please verify the object exists ,and is online, then try again.'
+
+        cluster = None
+        connected_esxi_hosts = []
+        esxi_hosts = []
+
+        compute_resources = self.create_compute_container_view()
+        for compute in compute_resources:
+            if compute.name == cluster_name:
+                cluster = compute
+
+        if cluster:
+            hgroup = datastore_utils.fa_hgroup_to_vc_cluster_mapping(cluster, flasharray)
+            esxi_hosts = cluster.host
+        else:
+            raise ValueError(f'Cluster \'{cluster_name}\' not found on specified vCenter Server. {msg}')
+
+        if esxi_hosts:
+            for esxi in esxi_hosts:
+                if esxi.runtime.connectionState == 'connected':
+                    connected_esxi_hosts.append(esxi)
+        else:
+            raise ValueError(f'No ESXi hosts found for \'{cluster_name}\'. {msg}')
+
+        return hgroup, connected_esxi_hosts
+
+
+class VmfsDatastores(Datastores):
+    """Maintenance of VMware VMFS datastores associated with Pure Storage FlashArrays."""
+
+    def __init__(self, vsphere_content, flasharray):
+        super(VvolDatastores, self).__init__(vsphere_content, flasharray)
+
+    def create_vmfs_datastore(self, cluster_name, ds_name, ds_size, flasharray, vmfs_version=None):
+        """Create a new VMFS datastore on a Pure Storage FlashArray and connect it to the requested vSphere cluster.
+
+        Args:
+            cluster_name (str): Name of the cluster to create datastore in.
+            ds_name (str): Requested name of the datastore.
+            ds_size (int): Size in GB the rquested datastore should be.
+            flasharray (purestorage.FlashArray): Existing session to FlashArray.
+            vmfs_version (int): Version of VMFS datastore to be created.
+
+        Returns:
+            datastore (vim.Datastore): Newly created datastore object.
+
+        Raises:
+            ValueError:
+                - If cluster resource is not found within vCenter.
+                - If no ESXi hosts are found (empty cluster).
+                - If no connected ESXi hosts are found within the cluster.
+                - If disk device is not found on ESXi host.
+        """
+        hgroup, connected_esxi_hosts = Datastores.verify_vsphere_cluster(cluster_name, flasharray)
```

Comment:
since this is not a static method I think super().verify_vsphere_cluster is the way to call this?

Reply:
Since I am not overriding it locally in the child class I do not need to use super(). I would only need to use super() if I overwrote it in my child class and wanted to use what the parent had for this particular call. 

-jhop